modules/retail/pages/retail-deploy-terminals-other.adoc

```diff
@@ -0,0 +1,158 @@
+. Prepare a USB device, and boot the terminal.
+For more information about using USB devices to boot, see xref:retail:retail-deploy-terminals-other.adoc[].
+. Alternatively, you can create the file and add it to the ``initrd``:
++
+----
+echo ./etc/sysconfig/network/ifcfg-wlan0 | cpio -H newc -o | gzip >> /srv/saltboot/boot/initrd.gz
+----
+
+Check that the terminals you want to deploy allow local booting.
+You can check this by editing the Saltboot formula in the {smr} {webui}.
+For more information about the Saltboot formula, see xref:salt:formula-saltboot.adoc[].
+
+=== Change Wireless Credentials
+
+After you have set the wireless credentials, you can change them as needed.
+The way to do this is different if you use one company-wide network, if you have each branch server on its own seperate network, or if you are using different wireless networks for booting and for regular system use.
+
+
+
+.Procedure: Changing Wireless Credentials for Single Network
+. Rebuild the boot image.
+. At the kernel command prompot, update the the [command]``WIRELESS_ESSID`` and [command]``WIRELESS_WPA_PSK`` kernel options.
+. Recreate the bootable USB device based on the new boot image.
+. Boot terminal from new USB device.
+
+
+
+.Procedure: Changing Wireless Credentials for Multiple Networks
+. In the [path]``/srv/salt/`` directory, create a salt state called ``update-terminal-credentials.sls``, and copy the details from [path]``/etc/sysconfig/network/ifcfg-wlan0``.
+At a command prompt on the terminal:
++
+----
+/etc/sysconfig/network/ifcfg-wlan0
+  file.managed:
+   - contents: |
+        WIRELESS_ESSID=<wireless_network_name>
+        WIRELESS_WPA_PSK=<wireless_network_password>
+# regenerate initrd
+  cmd.run:
+  - name: 'mkinitrd'
+----
+. Apply the Salt state to the terminal:
++
+----
+salt <terminal_salt_name> state.apply update-terminal-credentials
+----
+
+[[NOTE]]
+====
+If you are using a separate network for the boot phase, the managed file might need to be renamed, or extended to [path]``/etc/sysconfig/network/initrd-ifcfg-wlan0``.
+====
+
+
+
+.Procedure: Changing Wireless Credentials for Seperate Boot Network
+. Write a custom Salt state named ``saltboot_hook`` to provide custom ``/etc/sysconfig/network/ifcfg-wlan0`` to the system image during boot:
++
+----
+{% set root = salt['environ.get']('NEWROOT') %}
+{{ root }}/etc/sysconfig/network/ifcfg-wlan0:
+  file.managed:
+    - source:
+      - <salt://any/file>
+    - require:
+      - saltboot: saltboot_fstab
+    - require_in:
+      - saltboot: boot_system
+----
+The hook is applied after the system image is deployed.
+. Create a different version for use during booting, called ``/etc/sysconfig/network/initrd-ifcfg-wlan0``.
+This file can be maintained by Salt on running terminals, to provide wireless credentials for use only during bootstrap and booting.
```

Comment:
This is not related to the state above, but that generally either during image building or when adding branch specific password, this file `/etc/sysconfig/network/initrd-ifcfg-wlan0` can be used to specify password for boot phase.

Reply:
So, is the first one (``/etc/sysconfig/network/ifcfg-wlan0``) for normal operation, and the second one (``/etc/sysconfig/network/initrd-ifcfg-wlan0``) for boot? I feel like I have these mixed up.

Reply:
Yes, you are correct. Keep in mind that this setup is optional, however. If there is no `/etc/sysconfig/network/initrd-ifcfg-wlan0`, then `/etc/sysconfig/network/ifcfg-wlan0` will be used for both boot and normal operation.

Reply:
Particularly for this snipped, the `/etc/sysconfig/network/initrd-ifcfg-wlan0` for boot cannot be effectively managed by salt using `saltboot_hook`. Boot wireless network must be already running for salt to start, and `saltboot_hook` state cannot permanently modify `initrd`.

Reply:
Got it. Can you take another look and see if I'm closer now?